python/lib/sift_client/types/asset.py

```diff
@@ -42,15 +42,10 @@ def created_by(self):
     def modified_by(self):
         raise NotImplementedError
 
-    @property
-    def runs(self):
-        # TODO: how to implement these efficiently without fetching all?
-        # Cached property?
-        raise NotImplementedError
-
-    @property
-    def channels(self):
-        raise NotImplementedError
+    def runs(self, limit: int | None = None):
+        if not hasattr(self, "client") or self.client is None:
+            raise RuntimeError("Asset was somehow created with no SiftClient. Cannot make calls.")
```

Comment:
Ahh thank you, this is more clear now.

I just noticed this, do you think this error might actually be redundant with this one? https://github.com/sift-stack/sift/blob/6cdb1176438d65aab43d5a84f1ab617a38337268/python/lib/sift_client/types/_base.py#L25-L27
(I wonder if it would be more accurate to make this an `AttributeError` as well)